TrailBlaze/View Controllers/HomeViewController.m

```diff
@@ -98,13 +99,20 @@ - (IBAction)didTapTrailRun:(id)sender {
         [_trailrunButton setTitle:@"END" forState:UIControlStateNormal];
         if (isReadyToStartRun) {
             self->isReadyToStartRun = false;
-            [Run uploadRun:currentRoute withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
-                if (succeeded) {
-                    NSLog(@"run sent!");
-                } else {
-                    NSLog(@"run not sent");
+            if (_isRerun) {
+                PaceImprovementTracking *pacer = [[PaceImprovementTracking alloc] initWithRunID:@"dfsadfsg"];
+                if ([PaceImprovementTracking isAtStartPosition:currentLocation firstPoint:pacer.polylinePoints[0]]) {
```

Comment:
we discussed having the fetch of pace data be async when you initialize the tracker. if you do this then you won't be able to immediately call `isAtStartPosition` right after init.

Reply:
To resolve this I have the query for the run first and the allocation of the pacer as well as the isAtStartPosition call inside the completion block of the run query. The info from the query is then used to populate the fields of the pacer that require data the run query.